src/pybind/mgr/cephadm/module.py

```diff
@@ -2533,6 +2533,19 @@ def _apply(self, spec: GenericSpec) -> str:
             # should only refresh if a change has been detected
             self._trigger_preview_refresh(specs=[cast(DriveGroupSpec, spec)])
 
+        if spec.service_type == 'prometheus':
+            spec = cast(PrometheusSpec, spec)
+            if spec.retention_time:
+                valid_units = ['y', 'w', 'd', 'h', 'm', 's']
+                m = re.search(rf"^(\d+)({'|'.join(valid_units)})$", spec.retention_time)
```

Comment:
can we unit test this function? I think we ought to have something small to make sure these regex we're using work as intended.

Reply:
Added extended UT for retention time and retention size